src/cargo/util/config/contexts.rs

```diff
@@ -0,0 +1,13 @@
+use crate::core::features::AllowFeatures;
+use crate::core::features::UnstableFeatureContext;
+use crate::Config;
+
+impl UnstableFeatureContext for Config {
```

Comment:
I'm not usually a fan of trait impls being separate from either the trait or the struct definition.

Reply:
(`annotate-snippets` took this to an extreme)